semver/stringify.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+import { ANY } from "./constants.ts";
+import type { SemVer } from "./types.ts";
+
+function formatNumber(value: number) {
+  if (value === Number.POSITIVE_INFINITY) {
+    return "∞";
+  } else if (value === Number.NEGATIVE_INFINITY) {
+    return "⧞";
+  } else {
+    return value.toFixed(0);
+  }
+}
+
+/**
+ * Format a SemVer object into a string.
+ *
+ * If any number is NaN then NaN will be printed.
+ *
+ * If any number is positive or negative infinity then '∞' or '⧞' will be printed instead.
+ *
+ * @param semver The semantic version to format
+ * @returns The string representation of a semantic version.
+ */
+export function stringify(semver: Partial<SemVer>) {
```

Comment:
Why support `Partial<SemVer>`? I think the API should only deal with full `Semver`

Reply:
The current `format` implementation supports partial stringification by passing a `FormatStyle`. `Partial<SemVer>` basically replaces the `FormatStyle`:

Before:
```ts
format({ major: 1, minor: 2, patch: 3, prerelease:[], build: [] }, "patch") // returns 3
format({ major: 1, minor: 2, patch: 3, prerelease:[], build: [] }, "primary") // returns 1.2.3
format({ major: 1, minor: 2, patch: 3, prerelease:["pre"], build: [] }, "release") // returns 1.2.3-pre
```
After: 
```ts
stringify({ patch: 3 }) // returns 3
stringify({ major: 1, minor: 2, patch: 3 }) // returns 1.2.3
stringify({ major: 1, minor: 2, patch: 3, prerelease: ["pre"] }) // returns 1.2.3-pre
```

Reply:
`{ patch: 3 }` is not a semver. I don't see the benefit of dealing with it. What is the use case of such cases?

On the other hand, `format` only deals with actual semvers. That looks more reasonable design to me.

Reply:
I cannot think of any use case because one might just call `String(version.patch)` or `${version.patch}` or `version.patch.toString()` for a single stringified value.

I think the main point of annoyance with `SemVer` as a type is the need to pass `prerelease` and `build` even if they are empty arrays, because they can be inferred.

I think there are two ways to go about this:

1. Changing the type to
```ts
export function stringify(semver: {
    major: number;
    minor: number;
    patch: number;
    prerelease?: (string | number)[];
    build?: string[];
  }) { ... }
```

or use `SemVer` but openin a PR that makes `prerelease` and `build` optional in general. That would also simplify other parts of the mod.

WDYT?

Reply:
> 2. Use SemVer but opening a PR that makes prerelease and build optional in general. That would also simplify other parts of the mod.

This makes sense to me.